developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -0,0 +1,2124 @@
+[[how-to-create-a-plugin]]
+= How to Create a Plugin
+:toc: right
+:toclevels: 5
+
+This goal of this tutorial is to help users quickly and easily create
+Foreman plugins. This is not an exhaustive tutorial of
+http://rubyonrails.org/[Ruby on Rails] or an explanation of how
+http://guides.rubyonrails.org/engines.html[rails engines] work.
+
+[[naming-your-plugin]]
+== Naming your plugin
+
+It's strongly recommended that your plugin begins with the string
+"foreman" to help people identify its relationship with the project.
+Plugins are published as gems to rubygems.org, so check that the name
+you wish to use is free - again, using a standard prefix helps. The
+prefix is also assumed in the installer, so makes adding support there
+easier.
+
+Multiple words in the gem name should be separated with underscores
+("_"), although some plugins use underscores for the gem name and
+hyphens for git repo names. The git repo name isn't as important, but
+it's suggested to keep them to same to help people find it.
+
+A good example is *foreman_hooks* because the name clearly states it's a
+foreman plugin that adds hooks.
+
+[[using-the-example-plugin]]
+== Using the example plugin
+
+
+There is a fully working example plugin which you can clone to get you
```

Comment:
I would drop the last `you` - which you can clone to get quickly started

Reply:
Done